src/python/gudhi/datasets/remote.py

```diff
@@ -0,0 +1,101 @@
+# This file is part of the Gudhi Library - https://gudhi.inria.fr/ - which is released under MIT.
+# See file LICENSE or go to https://gudhi.inria.fr/licensing/ for full license details.
+# Author(s):       Hind Montassif
+#
+# Copyright (C) 2021 Inria
+#
+# Modification(s):
+#   - YYYY/MM Author: Description of the modification
+
+import hashlib
+
+from os.path import join, exists
+from os import makedirs
+
+from urllib.request import urlretrieve
+
+
+def _checksum_sha256(file_path):
+    """
+    Compute the file checksum using sha256
+
+    Parameters
+    ----------
+    file_path: string
+        Full path of the created file.
+
+    Returns
+    -------
+        The hex digest of file_path
+    """
+    sha256_hash = hashlib.sha256()
+    chunk_size = 4096
+    with open(file_path,"rb") as f:
+        # Read and update hash string value in blocks of 4K
+        while True:
+            buffer = f.read(chunk_size)
+            if not buffer:
+                break
+            sha256_hash.update(buffer)
+    return sha256_hash.hexdigest()
+
+def fetch(url, filename, dirname = "remote_datasets", file_checksum = None):
+    """
+    Fetch the wanted dataset from the given url and save it in file_path
+
+    Parameters
+    ----------
+    url : string
+        The url to fetch the dataset from.
+    filename : string
+        The name to give to downloaded file.
+    dirname : string
+        The directory to save the file to. Default is "remote_datasets".
+    file_checksum : string
+        The file checksum using sha256 to check against the one computed on the downloaded file.
+        Default is 'None'.
+
+    Returns
+    -------
+    file_path: string
+        Full path of the created file.
+    """
+
+    file_path = join(dirname, filename)
+
+    # Check that an existing file does not already exist at file_path
+    if not exists(file_path):
```

Comment:
I wonder if we should still verify the checksum, in case the file is just a truncated version left over from an interrupted previous download for instance. Or maybe urllib has enough guarantees that this isn't useful...

Reply:
It depends if fetch is made private or not; if public and checksum is not provided I would say that verifying the checksum of the found local file would be good if the user provides it for a second (or more) fetch call. Or, we can also make the checksum mandatory ... In that case, it would be enough to verify it only the first time (unless the user downloads the file manually (or by other means other than fetch) and the download fails without him noticing and then uses fetch another time).

Reply:
Here, image default checksum is `''` or `None`, I would do:
```python
remote._fetch_remote("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/spiral_2d/spiral_2d.npy", "remote_datasets/spiral_2d")
# Error: checksum '' is different from '88312ffd6df2e2cb2bde9c0e1f962d7d644c6f58dc369c7b377b298dacdc4eaf'
```
Like you can copy/paste and relaunch the correct command `remote._fetch_remote("https://raw.githubusercontent.com/GUDHI/gudhi-data/main/points/spiral_2d/spiral_2d.npy", "remote_datasets/spiral_2d", file_check_sum = '88312ffd6df2e2cb2bde9c0e1f962d7d644c6f58dc369c7b377b298dacdc4eaf')`

I have no strong opinion wether this method should be public or private.